src/com/google/enterprise/adaptor/ContentTransformFactory.java

```diff
@@ -0,0 +1,97 @@
+package com.google.enterprise.adaptor;
+
+import com.google.common.base.Strings;
+
+import java.io.OutputStream;
+import java.lang.reflect.Constructor;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * The content transform factory holds all document content transforms
+ * and puts them in series connection.
+ *
+ * @author Dominik Weidenfeld (dominik.weidenfeld@twt.de)
+ */
+public class ContentTransformFactory {
+
+  private Map<Constructor<DocumentContentTransform>, Map<String, String>>
+      transforms;
+
+  /**
+   * Constructs a new ContentTransformFactory. Also checks the basic
+   * configuration values for the {@link DocumentContentTransform}
+   * instantiation.
+   *
+   * @param transforms Configuration for each {@link DocumentContentTransform}
+   * @throws InvalidConfigurationException If the class for a {@link
+   *                                       DocumentContentTransform} is missing
+   * @throws RuntimeException              If the class does not match all
+   *                                       criteria
+   */
+  public ContentTransformFactory(
+      final List<Map<String, String>> transforms) {
```

Comment:
although it's common for ctors to use the same name of param and inner class variable, it's atypical and a bit confusing to do so in this method.  Please change parameter.  Perhaps change transforms to configs or confugations.